### ISSUE TITLE
comment_code_linter's false positives single quotes enclose strings on roxygen comments

### ISSUE THREAD

**Issue by eitsupi** [r-lib/lintr#1908]
It seems that lintr complains about the first line of the following R file.

```r
#' foo <- '
#' bar'
#' foo <- "
#' bar"
```

I found this in apache/arrow#33968

**Comment by IndrajeetPatil** [comment]
Thanks. Can reproduce:

``` r
library(lintr)

lint(
  text = "#' foo <- '
    #' bar'",
  linters = commented_code_linter()
)
#> <text>:1:2: style: [commented_code_linter] Commented code should be removed.
#> #' foo <- '
#>  ^~~~~~~~~~
```

<sup>Created on 2023-02-02 with [reprex v2.0.2](https://reprex.tidyverse.org)</sup>

**Comment by MichaelChirico** [comment]
I think this is downstream of making lintr roxygen-aware. once that's done I imagine this issue will go away on its own.

**Comment by MichaelChirico** [comment]
@AshesITR WDYT about completely excluding roxygen2 comments from `commented_code_linter()`? As simple as `//COMMENT[not(starts-with(text(), "#'"))]` instead of all `//COMMENT` nodes.
